homeassistant/components/zwave_js/__init__.py

```diff
@@ -573,6 +572,26 @@ def register_node_in_dev_reg(self, node: ZwaveNode) -> dr.DeviceEntry:
         else:
             ids = {device_id}
 
+        # Check if this node was previously provisioned
+        provisioning_entry = await controller.async_get_provisioning_entry(node.node_id)
+
+        if (
+            provisioning_entry
+            and hasattr(provisioning_entry, "device_id")
+            and provisioning_entry.device_id
+        ):
+            preprovisioned_device = self.dev_reg.async_get(provisioning_entry.device_id)
+
+            if preprovisioned_device:
+                dsk = provisioning_entry.dsk
+                dsk_identifier = (DOMAIN, f"provision_{dsk}")
+
+                # If the pre-provisioned device has the DSK identifier, update it
+                if dsk_identifier in preprovisioned_device.identifiers:
+                    self.dev_reg.async_update_device(
+                        preprovisioned_device.id, new_identifiers=ids
```

Comment:
We shouldn't update an existing device and replace all the identifiers with just our identifiers. A device may have other identifiers that we don't manage or know about and shouldn't touch. We need to copy the existing identifiers and remove the stale identifier that we want to replace and then add the new identifier to the copied identifiers before we update the device with the new identifiers. See example above on lines 566-570.